src/pydicom/pixels/decoders/native.py

```diff
@@ -132,19 +137,49 @@ def _rle_decode_frame(
     nr_segments = len(offsets)
 
     # Check that the actual number of segments is as expected
-    bytes_per_sample = nr_bits // 8
-    if nr_segments != nr_samples * bytes_per_sample:
+    if nr_bits == 1:
+        if nr_samples != 1:
+            raise ValueError(
+                "Cannot decode RLE pixel data with Bits Allocated = 1 "
+                "and SamplesPerPixel > 1."
+            )
+
+        bytes_per_sample = 1
+        expected_nr_segments = 1
+
+        # Since there can only be one segment, the length of the segment, the
+        # stride and total decoded length are all the same, and equal to the
+        # decoded length n bytes of the bit-packed array
+        decoded_segment_length = (rows * columns - 1) // 8 + 1
```

Comment:
Might be easier to understand with just do `math.ceil(rows * columns / 8)`

Reply:
resolved in [c1e9a0f](https://github.com/pydicom/pydicom/pull/2216/commits/c1e9a0f2c1b8225b608765a11be870fdd5d7fc99)